admin/securedrop_admin/__init__.py

```diff
@@ -690,6 +699,45 @@ def setup_logger(verbose: bool = False) -> None:
     sdlog.addHandler(stdout)
 
 
+def update_check_required(cmd_name: str) -> Callable[[_FuncT], _FuncT]:
+    """
+    This decorator can be added to any subcommand that is part of securedrop-admin
+    via `@update_check_required("name_of_subcommand")`. It forces a check for
+    updates, and aborts if the locally installed code is out of date. It should
+    be generally added to all subcommands that make modifications on the
+    server or on the Admin Workstation.
+
+    The user can override this check by specifying the --force argument before
+    any subcommand.
+    """
+    def decorator_update_check(func: _FuncT) -> _FuncT:
+        @functools.wraps(func)
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
+            cli_args = args[0]
+            if cli_args.force:
+                sdlog.info("Skipping update check because --force argument was provided.")
+                return func(*args, **kwargs)
+
+            update_status, latest_tag = check_for_updates(cli_args)
+            if update_status is True:
+                sdlog.error("You are not running the most recent signed SecureDrop release "
+                            "on this workstation.")
+                sdlog.error("Latest available version: {}".format(latest_tag))
```

Comment:
I wonder if we can also print the current branch/tag the repository is in.

Reply:
I played with that as well in an earlier version. The problem is that the `current_tag` variable is only useful for version comparisons, as it produces highly confusing output e.g., on `develop`, due to its reliance on `git describe`. For current `develop`, that results in `0.6-rc2-2646-g5eaad3b04` (!), as that's the most recent reachable tag. In addition to a branch or tag, the local repo may be in `HEAD detached` state, or in any of a number of different git states (dirty repo, rebase in progress, etc.).

The first line of `git status` is generally informative, as is the starred line of `git branch`. We could collect either line (we should probably do so in `check_for_updates`) and print it out to ease debugging. I think that would be helpful. 

I have a slight preference for `git branch` as its output tends to be a bit more terse. I would suggest doing this in a follow-up PR, to keep this one tightly scoped to the update check. Here's a one-liner for doing that in Python (without error handling):

```python
re.search(r"\* (.*)\n", subprocess.check_output(['git', 'branch']).decode('utf-8')).group(1)
```

Reply:
I have no strong feelings one way or the other if the goal is just to print out the current branch/tag - but! the output from `git status` will also let you know if there are code changes in the repo (e.g. unsanctioned poking at ansible defaults), giving you a few more options:

- `you're on x branch/tag, update to <latest>`
- `you're on x branch/tag and there are changed files, resolve this then upgrade to latest`

This is probably not going to be a one-liner though.

Reply:
For now pushed to a WIP branch: c9d7ffb4d1f353ac5059fe08554968eef983415c

This tries to get us the best of both worlds:
- use `git branch` for terse information that can be included in the output
- recommends using `git status` for troubleshooting

Will still need its own test and may fail CI as-is. If the approach seems sound, happy to include in this PR, or in a follow-up PR, at the reviewer's discretion.

Reply:
Since review is still pending, I'll just clean this change up a bit further and fold it into this PR now.

Reply:
Done, and test plan updated. I've refrained from any further mangling of the `git branch` output as it's possible to use Tails in languages other than English.